tardis/io/atom_data/base.py

```diff
@@ -322,24 +292,40 @@ def __init__(
         self.atom_data = atom_data
         self.ionization_data = ionization_data
         self.levels = levels
-        # Not sure why this is need - WEK 17 Sep 2023
+        # Cast to float so that Numba can use the values in numpy functions
         self.levels.energy = self.levels.energy.astype(np.float64)
         self.lines = lines
 
+        collected_macro_atom_data = MacroAtomData(
+            macro_atom_data, macro_atom_references
+        )
+
         # Rename these (drop "_all") when `prepare_atom_data` is removed!
-        self.macro_atom_data_all = macro_atom_data
-        self.macro_atom_references_all = macro_atom_references
+        self.macro_atom_data_all = (
+            collected_macro_atom_data.transition_probability_data
+        )
+        self.macro_atom_references_all = (
+            collected_macro_atom_data.block_reference_data
+        )
 
         self.zeta_data = zeta_data
 
-        self.collision_data = collision_data
-        self.collision_data_temperatures = collision_data_temperatures
+        chianti_collision_data = ChiantiCollisionData(
+            collision_data, collision_data_temperatures
+        )
+
+        cmfgen_collision_data = CMFGENCollisionData(
+            yg_data, collision_data_temperatures
```

Comment:
I don't think that is quite right. I'm pretty sure the yg_data stores it as part of the columns. Or ChiantiCollisionData. have a quick look in the plasma

Reply:
Looking at the plasma `YgData` class, it uses both `yg_data` and `collision_data_temperatures`. See https://github.com/andrewfullard/tardis/blob/atom-data-refactor-v2/tardis/plasma/properties/atomic.py#L676 and https://github.com/andrewfullard/tardis/blob/atom-data-refactor-v2/tardis/plasma/properties/atomic.py#L687. For the Chianti-source collision_data, it is used in NLTEData, see https://github.com/andrewfullard/tardis/blob/atom-data-refactor-v2/tardis/io/atom_data/nlte_data.py#L18-L19 and https://github.com/andrewfullard/tardis/blob/atom-data-refactor-v2/tardis/io/atom_data/nlte_data.py#L86. 

Oddly in the `from_hdf` method in this class, `collision_data` is set to "dummy value" if the data is sourced from CMFGen, see https://github.com/andrewfullard/tardis/blob/atom-data-refactor-v2/tardis/io/atom_data/base.py#L199. That is why I did not include it in the CMFGen dataclass.